packages/fiber/src/core/renderer.ts

```diff
@@ -314,68 +353,61 @@ function createRenderer<TCanvas>(roots: Map<TCanvas, Root>, getEventPriority?: (
       // This will make sure events are only added once to the central container
       const localState = (instance.__r3f ?? {}) as LocalState
       if (instance.raycast && localState.handlers && localState.eventCount) {
         instance.__r3f.root.getState().internal.interaction.push(instance as unknown as THREE.Object3D)
       }
     },
-    getPublicInstance: (instance: Instance) => instance,
-    shouldDeprioritizeSubtree: () => false,
+    getPublicInstance: (instance) => instance,
     prepareForCommit: () => null,
-    preparePortalMount: (container: UseBoundStore<RootState>) => prepare(container.getState().scene),
+    preparePortalMount: (container) => prepare(container.getState().scene),
     resetAfterCommit: () => {},
     shouldSetTextContent: () => false,
     clearContainer: () => false,
-    detachDeletedInstance: () => {},
-    hideInstance(instance: Instance) {
+    hideInstance(instance) {
       // Deatch while the instance is hidden
       const { attach: type, parent } = instance?.__r3f ?? {}
       if (type && parent) detach(parent, instance, type)
       if (instance.isObject3D) instance.visible = false
       invalidateInstance(instance)
     },
-    unhideInstance(instance: Instance, props: InstanceProps) {
+    unhideInstance(instance, props) {
       // Re-attach when the instance is unhidden
       const { attach: type, parent } = instance?.__r3f ?? {}
       if (type && parent) attach(parent, instance, type)
       if ((instance.isObject3D && props.visible == null) || props.visible) instance.visible = true
       invalidateInstance(instance)
     },
-    createTextInstance: () => {},
+    createTextInstance: () => {
+      throw new Error('Text is not allowed in the R3F tree.')
+    },
     hideTextInstance: () => {
       throw new Error('Text is not allowed in the R3F tree.')
     },
     unhideTextInstance: () => {},
-    getCurrentEventPriority: () => (getEventPriority ? getEventPriority() : DefaultEventPriority),
+    // https://github.com/pmndrs/react-three-fiber/pull/2360#discussion_r916356874
     // @ts-ignore
+    getCurrentEventPriority: () => (_getEventPriority ? _getEventPriority() : DefaultEventPriority),
+    beforeActiveInstanceBlur: () => {},
+    afterActiveInstanceBlur: () => {},
+    detachDeletedInstance: () => {},
     now:
       typeof performance !== 'undefined' && is.fun(performance.now)
         ? performance.now
         : is.fun(Date.now)
         ? Date.now
-        : undefined,
-    // @ts-ignore
-    scheduleTimeout: is.fun(setTimeout) ? setTimeout : undefined,
-    // @ts-ignore
-    cancelTimeout: is.fun(clearTimeout) ? clearTimeout : undefined,
-    setTimeout: is.fun(setTimeout) ? setTimeout : undefined,
-    clearTimeout: is.fun(clearTimeout) ? clearTimeout : undefined,
+        : () => 0,
+    scheduleTimeout: () => (is.fun(setTimeout) ? setTimeout : undefined),
+    cancelTimeout: () => (is.fun(clearTimeout) ? clearTimeout : undefined),
```

Comment:
We'll want to pass timeout methods here directly, react will call these immediately, and expect `HostConfig['timeoutHandle']` to come from `setTimeout` or a proxy of it.

```jsx
scheduleTimeout: is.fun(setTimeout) ? setTimeout : undefined,
cancelTimeout: is.fun(clearTimeout) ? clearTimeout : undefined,
```